tests/test_utils.py

```diff
@@ -346,3 +347,61 @@ def test_get_compile_command_sort_args(tmpdir_cwd):
             "--no-annotate --no-emit-index-url --no-emit-trusted-host "
             "requirements.in setup.py"
         )
+
+
+@pytest.mark.parametrize(
+    ("values", "key"),
+    (
+        pytest.param(
+            ("foo", "bar", "baz", "qux", "quux"),
+            lambda s: s[0],
+            id="with key function",
+        ),
+        pytest.param(
+            (("f", "foo"), ("b", "bar"), ("b", "baz"), ("q", "qux"), ("q", "quux")),
+            None,
+            id="without key function",
+        ),
+        pytest.param(
+            iter(("foo", "bar", "baz", "qux", "quux")),
+            lambda s: s[0],
+            id="values as iterator with key function",
+        ),
+        pytest.param(
+            iter(
+                (("f", "foo"), ("b", "bar"), ("b", "baz"), ("q", "qux"), ("q", "quux"))
+            ),
+            None,
+            id="values as iterator without key function",
+        ),
+    ),
+)
+def test_lookup_table(values, key):
+    expected = {"b": {"bar", "baz"}, "f": {"foo"}, "q": {"quux", "qux"}}
+    assert lookup_table(values, key) == expected
+
+
+@pytest.mark.parametrize(
+    "values",
+    (
+        pytest.param(("foo", "bar", "baz"), id="values are not tuples"),
+        pytest.param((("f", "foo"), "b"), id="one of the values is not a tuple"),
+    ),
+)
+def test_lookup_table_requires_key(values):
+    with pytest.raises(
+        ValueError,
+        match=r"^The `key` function must be specified when the `values` are not empty\.$",
+    ):
+        assert lookup_table(values)
```

Comment:
When you expect it to raise an error, `assert` will never happen.
```suggestion
        lookup_table(values)
```

Reply:
Nice catch! Thanks. Fixed in 080f85.